### DISCUSSION TITLE
Using MapStruct in my project is giving me issue when i am build the project

### DISCUSSION THREAD

**Discussion by hameer32** [mapstruct/mapstruct#2965] [Q&A]
```java
@Mapper(componentModel = "spring")
public interface CustomerMapper {
    //Below Methods Used to convert DTO to ENTITY
    Customer convertDtoToEntity(CustomerDto dto);
    default Map<String, Group> listGroupToMap(List<CustomerDto.Group> value) {
      return value.stream().collect(Collectors.toMap(CustomerDto.Group::getGroupId,
          this::groupMapperDtoToEntity));
    }
    Group groupMapperDtoToEntity(CustomerDto.Group group);
    default Map<String, Band> mapBand(List<Band> value) {
      return value.stream().collect(Collectors.toMap(Band::getObjectId, band -> band));
    }
    //Below Methods Used to convert ENTITY to DTO
    CustomerDto convertEntityToDto(Customer Customer);
    default List<CustomerDto.Group> mapGroupToList(Map<String, Group> value) {
      List<Group> es = new ArrayList<>(value.values());
      return es.stream().map(this::groupMapperEntityToDto).collect(Collectors.toList());
    }
    CustomerDto.Group groupMapperEntityToDto(Group group);
    default List<Band> listBandFromMap(Map<String, Band> value) {
      return new ArrayList<>(value.values());
    }
  }
```
This is the common class: 
```java
@Data
public class Band {
    private String objectId;
  private String group;
  private EnumType enum;

  public enum EnumType {
    WITHOUT, INNER, OUTER, BOTH
  }
}
```

This is DTO: 
```java
@Data
public class CustomerDto {
  private String customerId;
  private List<Group> groups = new ArrayList<>();

  @NoArgsConstructor
  @AllArgsConstructor
  @Data
  public static class Group {
    private String groupId;
    private String groupName;
    private GroupType groupType;
    private List<Band> bands = new ArrayList<>();
  }
}
```

This is Entity:
```java
@Data
public class Customer {
  private String customerId;
  private Map<String, Group> groups = new HashMap<>();

  @NoArgsConstructor
  @AllArgsConstructor
  @Data
  public static class Group {
    private String groupId;
    private String groupName;
    private GroupType groupType;
    private Map<String, Band> bands = new HashMap<>();
  }
  public enum GroupType {GROUP1,GROUP2,GROUP3}

}
```

Build.gradle 👍🏻 : 
```gradle
implementation 'org.mapstruct:mapstruct:1.5.2.Final'
annotationProcessor 'org.mapstruct:mapstruct-processor:1.5.2.Final'
```

Error What i am getting

<img width="853" alt="Screenshot 2022-08-19 at 6 32 42 PM" src="https://user-images.githubusercontent.com/33780351/185624862-9ea1f4da-407c-4b02-88d2-75f32d491a19.png">
<img width="1127" alt="Screenshot 2022-08-19 at 6 35 34 PM" src="https://user-images.githubusercontent.com/33780351/185624892-ad2231ba-4dfd-49d2-bf6f-d3dcede00faa.png">
<img width="663" alt="Screenshot 2022-08-19 at 6 35 57 PM" src="https://user-images.githubusercontent.com/33780351/185624905-07966d68-f2c4-4022-9872-5b1fff034e77.png">

**Comment by hameer32** [comment]
After removing mapstruct dependency and that mapper annotation , and directly adding that implemented class of the mapstruct interface. build is working

**Reply by filiphr** [reply]
I can't see any problem with MapStruct for this. I've delomboked your classes and tried and everything worked fine.

One thing I noticed is the fact that the name of the `EnumType` field in the `Band` is `enum`, which isn't a valid identifier. I changed this to `enumType` and the code compiled.

**Reply by hameer32** [reply]
That EnumType is original variable replacment . 
I have also used mapstruct in otger project, never faced this issue.
Actually this take a lot of time to build , earlier when starting spring boot app in local i used to get this error, after copying the project and building that worked fine. So i deleted the original one.
We can connect if you wanna see the original place of use for this.

**Reply by filiphr** [reply]
The strange thing is that in the stacktrace there are Lombok packages and not MapStruct. Perhaps you have something corrupted in your project. I would suggest trying a fresh clone in a new folder
